contributors/design-proposals/bootstrap-discovery.md

```diff
@@ -55,31 +55,35 @@ Now that we know *what* we want to get to the client, the question is how.  We w
 
 ### Method: Out of Band
 
-The simplest way to do this would be to simply put this object in a file and copy it around.  This is more overhead for the user, but it is easy to implement and lets users rely on existing systems to distribute configuration.
+The simplest way to obtain ClusterInfo this would be to simply put this object in a file and copy it around.  This is more overhead for the user, but it is easy to implement and lets users rely on existing systems to distribute configuration.
 
 For the `kubeadm` flow, the command line might look like:
 
 ```
-kubeadm join --cluster-info-file=my-cluster.yaml
+kubeadm join --discovery-file=my-cluster.yaml
 ```
 
-Note that TLS bootstrap (which establishes a way for a client to authenticate itself to the server) is a separate issue and has its own set of methods.  This command line may have a TLS bootstrap token (or config file) on the command line also.
+After loading the ClusterInfo from a file, the client MAY look for updated information from the server by reading the `kube-public` `cluster-info` ConfigMap defined below.  However, when retrieving this ConfigMap the client MUST validate the certificate chain when talking to the API server.
+
+**Note:** TLS bootstrap (which establishes a way for a client to authenticate itself to the server) is a separate issue and has its own set of methods.  This command line may have a TLS bootstrap token (or config file) on the command line also.  For this reason, even thought the `--discovery-file` argument is in the form of a `kubeconfig`, it MUST NOT contain client credentials as defined above.
 
 ### Method: HTTPS Endpoint
 
-If the ClusterInfo information is hosted in a trusted place via HTTPS you can just request it that way.  This will use the root certificates that are installed on the system.  It may or may not be appropriate based on the user's constraints.
+If the ClusterInfo information is hosted in a trusted place via HTTPS you can just request it that way.  This will use the root certificates that are installed on the system.  It may or may not be appropriate based on the user's constraints.  This method MUST use HTTPS.  Also, even though the payload for this URL is the `kubeconfig` format, it MUST NOT contain client credentials.
```

Comment:
I think you misunderstand.

If you already have credentials and a kubeconfig you don't need to use the bootstrap mechanism at all.  We'll be able to skip that phase.

But if you *don't* already have a kubeconfig kubeadm supports the following flow:
1) Discovery where the cluster is and what root CA bundles to use
2) Use some temporary credentials to submit a CSR to the cert API

This is about how we do those.  One method of doing (1) above is to hit a well known https endpoint.  In that case, if users accidentally upload a kubeconfig with credentials then they are exposing those more widely than they should.